web/src/components/layers/solar.js

```diff
@@ -102,20 +102,20 @@ class SolarLayer {
 
   show() {
     if (this.canvas) {
-      d3.select(this.canvas)
-        .style('display', 'block')
-        .transition().style('opacity', 1);
+      this.canvas.style.display = 'block';
+      this.canvas.style.opacity = 1;
     }
     this.hidden = false;
     this.render();
   }
 
   hide() {
+    if (this.hidden) return;
     if (this.canvas) {
-      d3.select(this.canvas).transition().style('opacity', 0)
-        .on('end', function() {
-          d3.select(this).style('display', 'none');
-        });
+      this.canvas.style.opacity = 0;
```

Comment:
any reason to remove the transition?

Reply:
Yes - due to the changes in polling logic from this PR, `hide()` and `show()` might now be called initially very soon after the app loads, sometimes before the layer is finished initializing, which would occasionally result in `d3.select(this.canvas).transition` being undefined, so I changed it here to CSS transitions ([see here](https://github.com/tmrowco/electricitymap-contrib/pull/2388/files/637bc3a7888cce7bdd9d9f070aceda52de314fed#diff-79e2defbaeeca7e8e55cd2a14fdb2f30R41)) which always work.

I agree the solution is kinda hacky, especially given the `setTimeout` below, but I'm going to start reworking this piece of code today as a part of https://github.com/tmrowco/electricitymap-contrib/issues/2310 so I think this should be an ok temporary compromise.

Reply:
ok thx!